Added filter for projects and tasks

snaplemouton opened the pull request:
| Question               | Answer
| ---------------------- | -
| Bugs correction ?      | yes/no
| New features ?         | yes/no
| Related issues         | #number

---

##### What have you changed  ?
...

##### How did you do that ?
...

RignonNoel commented on tasklist/src/pages/listTask/listTask.html:
```diff
@@ -18,6 +18,38 @@
+    <ion-label>Status</ion-label>
+    <ion-select [(ngModel)]="searchDone" (ionChange)="getDataFromApi()">
+      <ion-option value="all" selected="true">All</ion-option>
+      <ion-option value="done">Done</ion-option>
```
It will be good to keep the same label for User Experience. The user only know `open` and `closed`

snaplemouton replied:
Changed label Done and Not done to Open and Closed

RignonNoel commented on tasklist/src/pages/listTask/listTask.html:
```diff
@@ -18,6 +18,38 @@
+    <ion-searchbar [(ngModel)]="searchName" (ionInput)="getDataFromApi()"></ion-searchbar>
+  </ion-item>
+  <ion-item>
+    <ion-label>Id</ion-label>
```
The user don't know the `ID` and can't read it. You need to put the `ID` in the `list` and the `detail` of the task if you want filter by `ID`

snaplemouton replied:
Added Id to the list and details view

RignonNoel commented on tasklist/src/pages/listTask/listTask.html:
```diff
@@ -18,6 +18,38 @@
+      <ion-option value="7">7</ion-option>
+      <ion-option value="8">8</ion-option>
+      <ion-option value="9">9</ion-option>
+      <ion-option value="10">10</ion-option>
```
It works for me but i thinks its better to just able to `Sort by ` in place of `filter by` for the priority. But it's good no problem with that.

snaplemouton replied:
I'll leave it as a filter by for now because I cannot think of a way to have the layout allow for an intuitive way to sort by parameter without clogging the UI with lot of junk

RignonNoel commented on tasklist/src/pages/listProject/listProject.html:
```diff
@@ -15,6 +15,13 @@
+    <ion-searchbar [(ngModel)]="searchTerm" (ionInput)="getDataFromApi()"></ion-searchbar>
+  </ion-item>
+  <ion-item>
+    <ion-label>Id</ion-label>
```
The user don't know the ID and can't read it. You need to put the ID in the list and the detail of the project if you want filter by ID

snaplemouton replied:
Added the ID to the view

snaplemouton commented:
Added filter by assigned

snaplemouton commented:
Fixed the undefined bug and changed the format for ID on the UI.
Am currently unable to find a solution for placing the Show filter button in the popover, will try finding a solution.
